lib/events.js

```diff
@@ -0,0 +1,70 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const Sink = require('fxa-notifier-aws').Sink;
+const P = require('./promise');
+
+const config = require('./config').root();
+const db = require('./db');
+const logger = require('./logging')('events');
+const workers = require('./workers');
+
+const HEX_STRING = require('./validate').hex;
+
+var fxaEvents;
+
+if (!config.events.region || !config.events.queueUrl) {
+  fxaEvents = {
+    start: function start() {
+      if (config.env === 'prod') {
+        throw new Error('config.events must be included in prod');
+      } else {
+        logger.warn('accountEvent.unconfigured');
+      }
+    }
+  };
+} else {
+  fxaEvents = new Sink(config.events.region, config.events.queueUrl);
+
+  fxaEvents.on('data', function (message) {
+    logger.verbose('data', message);
+    if (message.event === 'delete') {
+      var userId = message.uid.split('@')[0];
+      if (!HEX_STRING.test(userId)) {
+        message.del();
+        return logger.warn('badDelete', { userId: userId });
+      }
+      P.all([
+        db.getAvatars(userId).then(function(avatars) {
+          return avatars.map(function(avatar) {
+            return workers.delete(avatar.id).then(function() {
```

Comment:
According to http://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/S3.html#deleteObject-property it won't error if there's no object.